sale_last_price_info/models/product.py

```diff
@@ -0,0 +1,29 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import api, fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    @api.one
+    def _get_last_sale(self):
+        """ Get last sale price, last sale date and last customer """
+        lines = self.env['sale.order.line'].search(
+            [('product_id', '=', self.id),
+             ('state', 'in', ['confirmed', 'done'])]).sorted(
+            key=lambda l: l.order_id.date_order, reverse=True)
```

Comment:
With thousands of sale lines it will start to be very ineffective I think. A direct `cr.execute()` followed by a `self.env['sale.order.line'].search([('id', 'in', ids)]` to apply the security rules could be something to consider.